src/main/java/com/parasoft/demoapp/service/GlobalPreferencesService.java

```diff
@@ -130,7 +133,13 @@ public GlobalPreferencesEntity updateGlobalPreferences(GlobalPreferencesDTO glob
 
         handleDemoBugs(currentPreferences, globalPreferencesDto);
 
-        handleRestEndpoints(currentPreferences, globalPreferencesDto);
+        WebServiceMode webServiceMode = globalPreferencesDto.getWebServiceMode();
```

Comment:
Rename handleRestEndpoints() to handleEndpoints(), and move these code into handleEndpoints(). Need to check null with webServiceMode.